src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -60,16 +90,71 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
-        }
-
+        }
+
+        public async Task<IEnumerable<ConfigurationClient>> GetAutoFailoverClients(CancellationToken cancellationToken)
+        {
+            var isUsingConnectionString = _connectionStrings != null && _connectionStrings.Any();
+
+            Uri endpoint = null;
+            var secret = string.Empty;
+            var id = string.Empty;
+
+            if (isUsingConnectionString)
+            {
+                var connectionString = _connectionStrings.First();
+
+                endpoint = new Uri(ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.EndpointSection));
+                secret = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.SecretSection);
+                id = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.IdSection);
+            }
+            else
+            {
+                endpoint = _endpoints.First();
+            }
+
+            IEnumerable<SrvRecord> results = await _srvLookupClient.Value.QueryAsync(endpoint.DnsSafeHost, cancellationToken).ConfigureAwait(false);
+
+            // shuffle the results to ensure hosts can be picked randomly.
+            IEnumerable<string> srvTargetHosts = results.Select(r => $"{r.Target.Value.Trim('.')}").Shuffle().ToList();
+
+            // clean up clients in case the corresponding replicas are removed.
+            foreach (var client in _clients)
+            {
+                if (IsEligibleToRemove(srvTargetHosts, client))
+                {
+                    _clients.Remove(client);
+                }
+            }
+
+            foreach (string host in srvTargetHosts)
+            {
+                if (!_clients.Any(c => c.Endpoint.Host.Equals(host, StringComparison.OrdinalIgnoreCase)))
+                {
+                    var targetEndpoint = new Uri($"https://{host}");
+
+                    var configClient = isUsingConnectionString? 
+                        new ConfigurationClient(ConnectionStringUtils.Build(targetEndpoint, id, secret), _clientOptions):
+                        new ConfigurationClient(targetEndpoint, _credential, _clientOptions);
+
+                    _clients.Add(new ConfigurationClientWrapper(targetEndpoint, configClient, true));
+                }
+            }
+
+            return _clients
+                .Where(client => client.IsAutoFailoverClient && 
+                    client.BackoffEndTime <= DateTimeOffset.UtcNow)
+                .Select(cw => cw.Client);
+        }
+
         public void UpdateClientStatus(ConfigurationClient client, bool successful)
         {
             if (client == null)
             {
                 throw new ArgumentNullException(nameof(client));
             }
 
-            ConfigurationClientWrapper clientWrapper = _clients.First(c => c.Client.Equals(client));
+            ConfigurationClientWrapper clientWrapper = _clients.FirstOrDefault(c => c.Client.Equals(client));
```

Comment:
This change introduced a null-ref.

Reply:
Indeed should not make this change, it's a lapse.